### PR TITLE
Refactor permissions checks (authorization) into single service

### LINKED ISSUE #664: DRY role index platform / compliance routes with `:type`

### LINKED ISSUE #680: Ember Service for auth abilities
There is a lot of scattered (and oft complex) logic with the new roles. @sandersonet and I discussed and Ember Service, a singleton, with descriptive long methods. Benefits:
- ease understanding / cognitive load
- much easier to unit test individual methods, than all the stubbing for acceptance scenarios
- all in one place
- clean

### PR DESCRIPTION
**Pull request by sandersonet**
The goal of this PR was to make it easier to detect the current user's status in the context of their organization.  Previously, this logic was scattered across many routes and components. `aptible-ability` offered some relief, but didn't work great outside of a template context.  `aptible-ability` also required passing down user context models (organization, roles, etc) down to low-level components that shouldn't care about these otherwise.

This PR centralizes the users entire context (and how they relate to all of their organizations) into a single authorization context service.  This sits on top of all authenticated routes and eagerly loads all resources required to make decisions about what UI a user should or shouldn't see.  Removing the async nature of loading all of these resources significantly reduces the complexity of many of our routes.

The service is injected to all routes, controllers, and components so signature bloat on any component that needs context in order to render should be greatly reduced.  (I fixed many of them, but there are probably more)

In addition to the authorization service refactor, this PR also adds a new "trainee dashboard" which shows the current users training history along with links to download their organizations latest policy manual.  See image at bottom of this PR.

Also included in this refactor

  * Opened it up so any organization member can click on any role to see members.  It was confusing that only some roles were clickable.  I also added logic to hide components that allowed editing of the role.  un-permissioned users can just see role members.
  * Consolidated role index pages (compliance | platform) into a single route with a type url param
  * Fixed role#show breadcrumb to role#index, it was previously broken
  * Added detection of current user's ability and redirect accordingly (on index pages)
  * Fixed A LOT of broken test stubs that were only working because we were over-stubbing--this made testing auth-related features much more difficult to reason about. 
  * The environment permission toggle on role#show was broken -- fixed
  * Add "My Gridiron" link to user dropdown
  * Nest "admin" organization screens in new parent route--makes it easier to enforce authorization
  * Renamed "compliance" => "gridiron" throughout
  * Separated "Apps and Databases" into new "enclave" route.  started using "enclave" throughout
  * Removed a lot of broken `activating-item` components in favor of a plain old `#link-to`.  We should remove this component entirely IMO. 

fixes #680
fixed #664

New "My Gridiron Status" page:
screenshot 2016-09-02 11 23 51

---
cc @gib

### CODE REVIEW

**Comment by sandersonet** [comment]
Some quick follow ups to this PR that I'd like to tackle soon:

* Refactor `aptible-ability` to use an `authorizationContext` to make decisions
* Finally make a damn application loading route ⏳

**Comment by sandersonet** [comment]
Forgot to mention -- with most resources now eagerly loaded (and duplicate requests squashed), Dashboard feels way fast again.

File: tests/acceptance/redirect-on-permission-test.js [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+      return this.success({ _embedded: { documents: [] }});
+    });
+  },
+
+  afterEach() {
+    Ember.run(application, 'destroy');
+  }
+});
+
+test('Users in `Owners` Role', function(assert) {
```
Comment (sandersonet) [inline tests/acceptance/redirect-on-permission-test.js]: The primary purpose of this PR was to make these test cases as easy as possible to detect and honor in dashboard.  Most other changes just precipitated from being able to support all of these use cases.

**Comment by gib** [comment]
Amazing! :bow: Thank you @sandersonet!!

File: app/components/activating-item/component.js [hunk 2]
```diff
@@ -15,6 +19,7 @@ export default Ember.Component.extend({
     return ALLOWED_PREFIXES.some((prefix) => {
       // NOTE: Perhaps we could have babel give us startsWith?
       let prefixedCurrentWhen = `${prefix}${currentWhen}`;
+      console.log(currentPath, currentWhen, prefixedCurrentWhen);
```
Comment (gib) [inline app/components/activating-item/component.js]: forgotten debugging `console.log`?

File: app/components/app-navigation/component.js [hunk 3]
```diff
@@ -3,43 +3,6 @@ import config from '../../config/environment';
...
-  opsActive: Ember.computed('routingService.currentPath', function() {
-    return !this.get('sheriffActive') && !this.get('settingsActive');
-  }),
-
-  settingsActive: Ember.computed('routingService.currentPath', function() {
-    let currentPath = this.get('routingService.currentPath');
-    return /members|contact|billing|settings|roles|invite|environments/.test(currentPath);
-  })
+  classNames: ['nav', 'nav-pills', 'application-nav'],
+  showSheriff: config.featureFlags['sheriff']
```
Comment (gib) [inline app/components/app-navigation/component.js]: 😍

File: app/components/app-navigation/template.hbs [hunk 4]
```diff
@@ -1,23 +1,24 @@
...
 {{/if}}
 
-{{#each currentUser.organizations as |organization|}}
-  <li class="{{if settingsActive 'active' null}}">
-    {{link-to organization.name 'organization' organization}}
-  </li>
+{{#each authorization.organizations as |organization|}}
+  {{#link-to 'organization' organization.id tagName="li"}}
+    {{link-to organization.name 'organization' organization.id class="organization-nav" bubbles=false}}
+  {{/link-to}}
```
Comment (gib) [inline app/components/app-navigation/template.hbs]: +100000 readability

File: tests/acceptance/redirect-on-permission-test.js [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+    assert.equal(find('.dashboard-dropdown-organization-menu .gridiron-user').length, 0,
+                'expectDenyMyGridiron: has no my gridiron link in dropdown nav');
+  });
+}
+
+function openUserToggle() {
+  let dropToggle = findWithAssert('.current-user .dropdown-toggle').eq(0);
+  dropToggle.click();
+}
+
```
Comment (gib) [inline tests/acceptance/redirect-on-permission-test.js]: This is amazing.

File: tests/acceptance/redirect-on-permission-test.js [hunk 6]
```diff
@@ -0,0 +1,508 @@
...
+  expectCanManageOrganizationAdmin(assert);
+
+  // Should be able to view "My Gridiron"
+  expectCanManageMyGridiron(assert);
+});
+
+test('Users ONLY in `Platform Owner` role', function(assert) {
+  // Enclave | Gridiron Admin | My Gridiron | Organization
+  // ---------------------------------------------------------------------------
+  // Manage  | Deny           | Manage      | Deny
```
Comment (gib) [inline tests/acceptance/redirect-on-permission-test.js]: Awesome comments / pattern

**Comment by sandersonet** [comment]
Updated to add an application loading route + removed the activated-item component entirely.

**Comment by gib** [comment]
Nice!

**Pull request merged**